### PR TITLE
FMA-25 Filter out any non-compliant Gaia-X credentials from the searc…

### PR DESCRIPTION
**Pull request by zoemaas**
…h results in the Federated Catalog

   Added SD compliance filter;

### CODE REVIEW

File: src/components/pages/Publish/Debug.tsx [hunk 1]
```diff
@@ -9,25 +9,30 @@ export default function Debug({
...
-      },
-      {
-        index: 1,
-        type: values.access,
-        serviceEndpoint: values.providerUri,
-        attributes: {}
-      }
-    ]
-  }
+  const [ddo, setDdo] = useState({})
```
Comment (nklomp) [inline src/components/pages/Publish/Debug.tsx:12]: What is the reason for this change?

File: src/components/templates/Search/utils.ts [hunk 2]
```diff
@@ -68,6 +69,12 @@ export function getSearchQuery(
 
   const nestedQuery = {
     must: [
+      {
+        query_string: {
+          default_field: 'service.attributes.additionalInformation.isCompliant',
+          query: !!isCompliant
```
Comment (nklomp) [inline src/components/templates/Search/utils.ts]: Probably compliantOnly is a better name for the filter.

Also I don't like boolean for these types I guess. How do i distinguish between only non-compliant, complaint and both? Let's leave it for now, given the amount of time we have.

File: content/site.json [hunk 3]
```diff
@@ -22,7 +22,7 @@
       },
       {
         "name": "Catalogue",
-        "link": "/search?sort=_score&sortOrder=desc&text="
+        "link": "/search?sort=_score&sortOrder=desc&isCompliant=true&text="
```
Comment (nklomp) [inline content/site.json]: Okay, where can the user update this filter?

**Review by nklomp — CHANGES_REQUESTED** [review]
Some small remarks. A big one is: where can the user change the filter?

File: src/components/pages/Publish/Debug.tsx (hunk 1 above)
Comment (zoemaas) [reply src/components/pages/Publish/Debug.tsx:12]: Following the existing pattern of the filters, `transformPublishFormToMetadata` is now asynchronous because I had to check whether the SD is in compliance and set the `isInCompliance` property  before I publish the asset. The compliance verification is async and happens inside of `transformPublishFormToMetadata` so I had to update every call to it.

File: src/components/templates/Search/utils.ts (hunk 2 above)
Comment (zoemaas) [reply src/components/templates/Search/utils.ts]: I already had some doubts about the naming, because I wanted to match it with the application patterns  and I wasn't sure if isInCompliance would be better than isCompliant.

File: content/site.json (hunk 3 above)
Comment (zoemaas) [reply content/site.json]: The user applies the filter clicking on a button. This change was a mistake and I already fixed it.

File: src/components/templates/Search/utils.ts (hunk 2 above)
Comment (zoemaas) [reply src/components/templates/Search/utils.ts]: I'm not sure if I understand what compliantOnly means...

File: src/utils/metadata.ts [hunk 4]
```diff
@@ -261,7 +261,20 @@ export function getInitialPublishFormDatasetsValues(
...
 }
 
-export function transformPublishFormToMetadata(
+const verifySelfDescription = async (args: {
+  url?: string
+  raw?: any
+}): Promise<boolean> => {
+  let result
+  if (args.url) {
+    result = await verifyServiceSelfDescription({ body: args.url, raw: false })
```
Comment (sksadjad) [inline src/utils/metadata.ts:270]: I think with the changes in the FMA-47 branch this filter should work correctly. we can test the filter functionality after merging that pr

**Review by sksadjad — CHANGES_REQUESTED** [review]
looks ok. please rebase it after merging FMA-47 and test it against the new 2210vp endpoints (already existing in that branch)

**Comment by zoemaas** [comment]
Code already present in `feature/FMA-47/vp-support`
